src/Illuminate/Support/Collection.php

```diff
@@ -1404,6 +1404,62 @@ public function prioritize( callable $condition ) {
 
 		return $this
 			->filter( $condition )
-			->union( $nonPrioritized );
+			->toBase()
+			->merge( $nonPrioritized );
+	}
+
+	/**
+	 * Convert an associative array of key => value to array of pairs [ key, value ].
+	 *
+	 * @return Collection
+	 */
+	public function assocToPair() {
+		return $this->keys()->zip( $this->values() );
+	}
```

Comment:
@BruceOnTheGo I find a bit hard to understand why it's written like that.

I had to check the documentation to understand what `zip` does.

Wouldn't the `map` method achieve the same result?

```
return $this->map( function ( $item, $key ) {
    return [ $key, $item ];
} )->values();
```

Reply:
@andreasciamanna Yes, it would achieve the same result but I prefer to use the provided functions when I can.

Reply:
@andreasciamanna I changed as you suggested because it should be more efficient.